src/lib/structures/Josh.ts

```diff
@@ -1694,7 +1710,7 @@ export class Josh<StoredValue = unknown> {
   }
 
   private resolvePath(path: Path): string[] {
-    return typeof path === 'string' ? path.split('.') : path;
+    return Array.isArray(path) ? path : path.replace(/\[/g, '').replace(/\]/g, '').split('.').filter(Boolean);
```

Comment:
```suggestion
    return Array.isArray(path) ? path : path.replace(/\[/g, '.').replace(/\]/g, '').split('.').filter(Boolean);
```
If you replace [ with nothing, it won't do the right thing :
`array[0].subprop` would become `["array0", "subprop"]`

Replace it with a point, but the ] doesn't need it as it is followed by one.

Reply:
Good catch. I probably should add a test for this.